### PR TITLE
Plane: implement quaternion based ACRO mode

### PR DESCRIPTION
**Pull request by tridge**
when ACRO_LOCKING=2 this implements a quaternion based ACRO controller, which allows for accurate rolls and knife edge

### CODE REVIEW

File: ArduPlane/Attitude.cpp [hunk 1]
```diff
@@ -471,6 +479,78 @@ void Plane::stabilize_acro(float speed_scaler)
...
+    error_angle2.z = constrain_float(error_angle2.z, -max_err_yaw_rad, max_err_yaw_rad);
+
+    // correct target based on max error
+    q.rotate_fast(error_angle2 - error_angle1);
+    q.normalize();
+
+    // convert to desired body rates
+    error_angle2.x /= rollController.tau();
+    error_angle2.y /= pitchController.tau();
+    error_angle2.z /= pitchController.tau(); // no yaw tau parameter, use pitch
```
Comment (IamPete1) [inline ArduPlane/Attitude.cpp:544]: This is a bit odd.

Comment (IamPete1) [reply ArduPlane/Attitude.cpp:544]: Maybe we just need to add one.

**Review by IamPete1 — COMMENTED** [review]
Looks like this brings us inline with how copter handles ACRO, there this method permanent locking is the default with a option to use rate loops only, that would be the equivalent of locking off I think. 

Would be nice to move to quaternions all the time, there could be lots of shared code (or at least shared maths) where plane could befit from all the work that has been done on copter. However I'm not really sure how we would deal with the lack of yaw control.

File: ArduPlane/Attitude.cpp (hunk 1 above)
Comment (tridge) [reply ArduPlane/Attitude.cpp:544]: very few people change the time constant on pitch, so adding a param just adds something nobody will change and has no basis to change. I'll add it if we find a reason to add it, ie. if we find a 3D aircraft that needs a different time constant on yaw and pitch

**Comment by andypnz** [comment]
I have not really used acro mode much before, but this looks to work really well in SITL....

**Review by andypnz — APPROVED** [review]

**Pull request merged**